### PR TITLE
[Yeelight] Use DS annotations, fix static code analysis findings

### PR DESCRIPTION
**Pull request by wborn**
* Use DS annotations
* Fix static code analysis findings
* Use diamond operator
* Replace Runnables with lambdas

### CODE REVIEW

File: addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java [hunk 1]
```diff
@@ -68,39 +67,36 @@ public boolean connect() {
...
-                            }
-                        } catch (Exception e) {
-                            logger.debug("Exception: {}", e);
+        mConnectThread = new Thread(() -> {
+            try {
+                mCmdRun = true;
+                logger.debug("{}: connect device! {}, {}", TAG, mDevice.getAddress(), mDevice.getPort());
+                mSocket = new Socket(mDevice.getAddress(), mDevice.getPort());
+                mSocket.setKeepAlive(true);
+                mWriter = new BufferedOutputStream(mSocket.getOutputStream());
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java]: I am not sure but shouldn't we maybe use a `try-with-resources` statement here (and the following line) to be on the safe side?

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java]: Even worse. Than we have to make sure it / they (`mWriter` and `mReader`) got closed in the `disconnect` method?

Comment (wborn) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java]: I don't see any other method calling the `disconnect` method. Maybe @claell knows how this should work?

Comment (claell) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java]: I know that I did not really understand the logic of this code the last time I got in touch with it. So I probably cannot be of help here unfortunately.
What I do know however is that it probably has some flaws in it, as it seems to have problems to catch up again after a device lost its connection during runtime (like turning it off and on again). Maybe this is related to problems with properly disconnecting a device?

**Comment by claell** [comment]
Thanks for the improvements, especially to the lib code!

File: addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java (hunk 1 above)
Comment (cweitkamp) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java]: Alright. For now I would say we leave this PR like it is and keep in mind that the connection handling could be improved some time in the future.

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java]: @wborn feel free to merge if you agree.

Comment (wborn) [reply addons/binding/org.openhab.binding.yeelight/src/main/java/org/openhab/binding/yeelight/internal/lib/device/connection/WifiConnection.java]: Yes the code can use some further improvements and fixes that can be done in follow up PRs.

**Pull request merged**